crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -948,9 +990,90 @@ mod tests {
                 log_dir: Default::default(),
             },
             "testDevice",
+            std::path::Path::new(""),
+            &mut std::collections::HashMap::default(),
         )
         .await
         .unwrap();
         assert_eq!(output, vec![]);
     }
+
+    /// Creates `n` child devices named "child-n".
+    /// for each child device a `k` is selected using a random seed so that
+    /// each child devices is assigned a random set of operations.
+    ///
+    /// The resulting dir structure is the following:
+    /// .
+    /// └── operations
+    ///     └── c8y
+    ///         ├── child-0
+    ///         │   └── c8y_LogfileRequest
+    ///         ├── child-1
+    ///         │   ├── c8y_Command
+    ///         │   ├── c8y_DownloadConfigFile
+    ///         │   └── c8y_SoftwareUpdate
+    ///         ├── child-2
+    ///         │   ├── c8y_Command
+    ///         │   ├── c8y_DownloadConfigFile
+    ///         │   └── c8y_LogfileRequest
+    ///         └── child-3
+    ///             └── c8y_LogfileRequest
+    fn make_n_child_devices_with_k_operations(n: u8, ttd: &TempTedgeDir) {
```

Comment:
Nice test utility. You've gone to quite some lengths to ensure that randomness across test runs.